astroquery/mast/core.py

```diff
@@ -1270,6 +1273,29 @@ def _download_curl_script(self, products, out_dir):
                           "URL": [url]})
         return manifest
 
+    def enable_s3_hst_dataset(self):
+        """
+        Attempts to enable downloading HST public files from S3 instead of MAST
+        """
+        try:
+            global boto3
```

Comment:
Why is this a global?  I think it should not be.  If anything, make the imported module into a semi-private attribute of the module (e.g., `self._boto3 = boto3`)